core/deployment/deployment_processor.go

```diff
@@ -69,3 +73,20 @@ func (d *deploymentProcessor) injectSupportingServiceEndpointIntoDeployment() er
 	}
 	return nil
 }
+
+func (d *deploymentProcessor) injectKogitoAnnotations() error {
+	if len(d.Context.Version) == 0 {
+		d.Log.Warn("Not able to get detect Kogito version, version annotation will not be set", "Version Label", framework.KogitoOperatorVersionAnnotation)
+	} else {
+		// annotate the given deployment with the operator version
+		if d.deployment.Annotations == nil {
```

Comment:
Why we decided to put the Annotations here and not in:
- Deployment.ObjectMeta.Annotations
- Deployment.Spec.Template.ObjectMeta.Annotations